### PR TITLE
[Components with more than 1 service] LPS-178823 Refactor DDMFormFieldTemplateContextContributor components with more than 1 service type

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-178823](https://issues.liferay.com/browse/LPS-178823)

### CODE REVIEW

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:forms

**Comment by zhangyan0701** [comment]
ci:stop

**Comment by zhangyan0701** [comment]
ci:test:forms

**Comment by Kvale1733** [comment]
@RenanUmV Regression found, form toggles that were not checked by default are no checked but don't show related fields until rechecked. This causes many Forms tests to fail since they depend on those fields being set when the the toggles are checked. I have attached videos to better show what I mean.

Steps to reproduce
1. Go to Forms admin in the produce menu
2. Go to add form
3. Add a text field
4. Observer Required field is not checked (when it is checked Error Message field should appear)
5. Go to Advanced Tab
6. Observe only Show Field is checked
7. Go to autocomplete tab
8. Observer Autocomplete is not checked (fields should appear when it is checked)

https://user-images.githubusercontent.com/90647688/226708246-4bb5637f-66e3-4e2b-9669-f2e64eb4aeea.mp4



https://user-images.githubusercontent.com/90647688/226708183-f97431bc-9675-4c28-ba9d-5cf9d86d46e5.mp4

**Comment by RenanUmV** [comment]
ci:test:forms

**Comment by Kvale1733** [comment]
@RenanUmV No regressions found.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/select/multi/language/option/MultiLanguageOptionSelectDDMFormFieldTemplateContextContributor.java [hunk 1]
```diff
@@ -75,8 +74,10 @@ availableLocale, getClass()),
 	@Reference
 	private Language _language;
 
-	@Reference
-	private SelectDDMFormFieldTemplateContextContributor
+	@Reference(
+		target = "(component.name=com.liferay.dynamic.data.mapping.form.field.type.internal.select.SelectDDMFormFieldTemplateContextContributor)"
+	)
```
[the remark this answers is not in the document]
Comment (rafaprax) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/select/multi/language/option/MultiLanguageOptionSelectDDMFormFieldTemplateContextContributor.java:79]: and put this change in another commit. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by rafaprax** [comment]
ci:test:sf

**Review by rafaprax — APPROVED** [review]

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by tinatian** [comment]
ci:forward

**Pull request closed without merging**